infra/dockerfile.base

```diff
@@ -1,17 +1,17 @@
 FROM python:3.10.7-slim
 
 # Setup dependencies in a cacheable step
-COPY requirements.txt /requirements.txt
-RUN apt-get update && \
-		apt-get install -y --no-install-recommends gcc g++ libgomp1 libffi-dev libjemalloc2 zstd patch && \
-		pip install --disable-pip-version-check --quiet --no-cache-dir -r /requirements.txt && \
-		apt-get purge -y gcc g++ libffi-dev patch && \
-		apt-get autoremove -y && \
-		rm -rf /var/lib/apt/lists/*
+RUN --mount=type=bind,source=requirements.txt,target=/requirements.txt,rw \
+	apt-get update && \
+	apt-get install -y --no-install-recommends gcc g++ libgomp1 libffi-dev libjemalloc2 zstd patch && \
+	pip install --disable-pip-version-check --quiet --no-cache-dir -r /requirements.txt && \
+	apt-get purge -y gcc g++ libffi-dev patch && \
+	apt-get autoremove -y && \
+	rm -rf /var/lib/apt/lists/*
 
 ENV LD_PRELOAD="libjemalloc.so.2"
 
 COPY infra/mozci_config.toml /root/.config/mozci/config.toml
 
-COPY . /tmp/bugbug
-RUN pip install --disable-pip-version-check --quiet --no-cache-dir /tmp/bugbug
+RUN --mount=type=bind,target=/tmp/bugbug,rw \
```

Comment:
```suggestion
RUN --mount=type=bind,target=/tmp/bugbug \
```

Reply:
Without `rw`, the build process fails with the following error:

```
5.767       error: could not create 'build': Read-only file system
```

According to this [thread](https://discuss.python.org/t/installing-from-a-read-only-source/4115), pip requires write access to build the wheel in the source directory. This issue persists unless you use `--use-deprecated=out-of-tree-build`, which is planned to be removed in pip version 22.1 as mentioned in this [GitHub pull request](https://github.com/pypa/pip/pull/10495/files).

Reply:
Thank you for the clarification.